### PR TITLE
fixed isFileBinaryGit

### PR DESCRIPTION
**Pull request by mullikine**

### CODE REVIEW

File: vcs/git/patch.go [hunk 1]
```diff
@@ -19,10 +19,7 @@ func isFileBinaryGit(filename string) (bool, error) {
 		return false, errors.Trace(err)
 	}
 
-	out, err := gitCMD("-C", repoRoot, "diff", "--no-index", "--numstat", "/dev/null", filename)
-	if err != nil {
-	    return false, errors.Trace(err)
-	}
+	out, _ := gitCMD("-C", repoRoot, "diff", "--no-index", "--numstat", "/dev/null", filename)
```
Comment (waigani) [inline vcs/git/patch.go]: Add a comment explaining why we're ignoring the error, as this is usually a "no no".

File: vcs/git/patch.go [hunk 2]
```diff
@@ -19,10 +19,8 @@ func isFileBinaryGit(filename string) (bool, error) {
 		return false, errors.Trace(err)
 	}
 
-	out, err := gitCMD("-C", repoRoot, "diff", "--no-index", "--numstat", "/dev/null", filename)
-	if err != nil {
-	    return false, errors.Trace(err)
-	}
+	// This command is allowed to fail. We only need to grep its output
+	out, _ := gitCMD("-C", repoRoot, "diff", "--no-index", "--numstat", "/dev/null", filename)
```
Comment (waigani) [inline vcs/git/patch.go]: It would be better to ignore only the expected error, if we know what that is.

Comment (mullikine) [reply vcs/git/patch.go]: The error code is '1'. I'm guessing that's the generic error code.

Comment (mullikine) [reply vcs/git/patch.go]: Ahh 1 means there were differences.

**Pull request merged**